### PR TITLE
feat: write binary files

### PR DESCRIPTION
**Pull request by MarcinVaadin**
## Description

New request support: write binary file

Fixes: #50

### CODE REVIEW

File: src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ import java.nio.channels.SocketChannel
 import java.util.*
 
 // Server implementation based on https://github.com/teocci/NioSocketCodeSample/tree/master
-@Deprecated("Will be removed in Copilot 24.5")
+@Deprecated("Will be removed in future")
```
Comment (Artur-) [inline src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt:16]: Seems unrelated

File: src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteBinaryFileHandler.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import com.intellij.psi.PsiManager
+import java.io.File
+import java.nio.file.Files
+import java.util.*
+
+class WriteBinaryFileHandler(project: Project, data: Map<String, Any>) : WriteFileHandler(project, data) {
+
+    override fun doWrite(vfsFile: VirtualFile?, doc: Document?, content: String) {
+        vfsFile?.let {
+            it.setBinaryContent(Base64.getDecoder().decode(content))
```
Comment (Artur-) [inline src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteBinaryFileHandler.kt:17]: What is the input - how does the command work? Is it binary or base64? If it is base64, maybe the method should be writeBase64 and not writeBinary?

Comment (MarcinVaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteBinaryFileHandler.kt:17]: It is base64 encoded binary data, to send pure binary data we would need post multipart request which requires lot more changes in api.

**Review by abdullahtellioglu — APPROVED** [review]
